services/src/main/java/org/keycloak/services/resources/LoginActionsService.java

```diff
@@ -412,17 +410,20 @@ public Response resetCredentialsPOST(@QueryParam(AUTH_SESSION_ID) String authSes
      * Endpoint for executing reset credentials flow.  If token is null, a authentication session is created with the account
      * service as the client.  Successful reset sends you to the account page.  Note, account service must be enabled.
      *
+     * @deprecated Please use endpoint '/realms/{realm_name}/protocol/openid-connect/forgot-credentials' for directly reset credentials from your client application
      * @param code
      * @param execution
      * @return
      */
     @Path(RESET_CREDENTIALS_PATH)
     @GET
+    @Deprecated
     public Response resetCredentialsGET(@QueryParam(AUTH_SESSION_ID) String authSessionId, // optional, can get from cookie instead
                                         @QueryParam(SESSION_CODE) String code,
                                         @QueryParam(Constants.EXECUTION) String execution,
                                         @QueryParam(Constants.CLIENT_ID) String clientId,
                                         @QueryParam(Constants.TAB_ID) String tabId) {
+        logger.warn("This endpoint is deprecated. Please use endpoint '/realms/{realm_name}/protocol/openid-connect/forgot-credentials' with parameters like OIDC for directly reset credentials from your client application");
```

Comment:
@mposolda This endpoint is the one that the `Forgot password` login option uses right now when it's shown in the login page. I think it cannot be changed to the OIDC one (the client initiating can be SAML for example). So I think we cannot deprecate it, we can say that it's only for internal use in the doc and move the warning when the auth session needs to be created (but I think that deprecation is not OK).